packages/@icon-magic/generate/src/plugins/svg-to-raster.ts

```diff
@@ -0,0 +1,122 @@
+ * break the plugin chain as a Flavor is returned, just that the content of the
+ * types are empty
+ */
+export const svgToRaster: GeneratePlugin = {
+  name: 'svg-to-raster',
+  iterants: ['sizes', 'resolutions'],
+  fn: async (flavor: Flavor, icon: Icon, params?: any): Promise<Flavor> => {
+    // get the size and resolution from the params passed in
+    if (params && params.propCombo) {
+      let w = params.propCombo.sizes.width || params.propCombo.sizes;
+      let h = params.propCombo.sizes.height || params.propCombo.sizes;
+      let res = params.propCombo.resolutions;
+
+      // get the generateOutputPath from the icon and make it if it doesn't exist
+      // alraedy
+      let outputPath = icon.generateOutputPath;
+      await fs.mkdirp(outputPath);
+
+      let realSize = `${w * res}x${h * res}`;
+      let assetName = `${flavor.name}-${realSize}`;
+
+      // First, we generate the png and store it in the output directory
+      let pngOutput = `${path.join(outputPath, assetName)}.png`;
+      debug(`Creating ${pngOutput}`);
+      await generatePng(
+        flavor.contents as string, // svg is always in a string format
+        w * res,
+        h * res,
+        pngOutput
+      );
+
+      // Convert the png to webp
+      debug(`Creating ${pngOutput} webp`);
+      let webpOut = await convertToWebp(
+        pngOutput,
+        `${path.join(outputPath, assetName)}.webp`
+      );
+
+      // minify both the png and webp assets
+      debug(`Minifying ${pngOutput} and .webp`);
+      await Promise.all([minify(pngOutput), minify(webpOut)]);
+
+      // create a new flavor with this sizexresolution combination
+      let flavorWithRasterAssets: Flavor = new Flavor(icon.iconPath, {
+        name: assetName,
+        path: `./${flavor.name}.svg`,
+        types: {
+          png: {
+            name: assetName,
+            path: `./${assetName}.png`
+          },
+          webp: {
+            name: assetName,
+            path: `./${assetName}.webp`
+          }
+        }
+      });
+      return flavorWithRasterAssets;
+    }
+  }
+};
+
+/**
+ * Generate a png asset from an svg string
+ * @param svg the contents of an svg file that needs to be converted to png
+ * @param width width of the output png
+ * @param height height of the output png
+ * @param outputPath path to write the png file after it's created
+ */
+async function generatePng(
+  svg: string,
+  width: number,
+  height: number,
+  outputPath: string
+): Promise<void> {
+  let png = await convert(svg, { width, height });
+  await fs.writeFile(outputPath, png);
+}
+
+/**
+ * Converts a png file to webp and writes to the outputPath
+ * @param input path to the input file in .png format
+ * @param outputPath path where the output needs to go
+ */
+function convertToWebp(pathToPng: string, outputPath: string): Promise<string> {
+  return new Promise<string>((resolve, reject) => {
+    webp.cwebp(pathToPng, outputPath, '-q 80', function(status: string) {
```

Comment:
Feel free to add a webp option to my svg to png farm now that puppiteer has a webp export option! Would be nice to do this in a single pass. Can you open an issue to add this as a fast-follow feature upgrade?

Reply:
I'm tracking it in JIRA right now, we should use github issues for it all, yes?